### PR TITLE
Rework signing, fix stateroots

### PR DESCRIPTION
**Pull request by roman-khimov**
Stateroots were not signed/verified properly, but there was also a leftover from #1741.

### CODE REVIEW

**Review by AnnaShaleva — CHANGES_REQUESTED** [review]

File: pkg/core/interop/crypto/ecdsa.go [hunk 1]
```diff
@@ -31,21 +31,20 @@ func ECDSASecp256r1CheckMultisig(ic *interop.Context) error {
 	if len(pkeys) < len(sigs) {
 		return errors.New("more signatures than there are keys")
 	}
-	sigok := vm.CheckMultisigPar(ic.VM, elliptic.P256(), hashToCheck.BytesBE(), pkeys, sigs)
+	sigok := vm.CheckMultisigPar(ic.VM, elliptic.P256(), hash.NetSha256(ic.Network, ic.Container).BytesBE(), pkeys, sigs)
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/core/interop/crypto/ecdsa.go:34]: I see that you got rid of all these hashes, so the question isn't relevant.

**Comment by AnnaShaleva** [comment]
neo-go's implementation of dbft's crypto.PrivateKey interface still uses an old hashing scheme, is it OK?

File: pkg/core/state/mpt_root.go [hunk 2]
```diff
@@ -9,6 +12,7 @@ import (
 
 // MPTRoot represents storage state root together with sign info.
 type MPTRoot struct {
+	Network netmode.Magic        `json:"-"`
 	Version byte                 `json:"version"`
 	Index   uint32               `json:"index"`
 	Root    util.Uint256         `json:"stateroot"`
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/core/state/mpt_root.go]: A bit different problem, but worth fixing, true.

File: pkg/services/oracle/request.go [hunk 3]
```diff
@@ -164,12 +164,12 @@ func (o *Oracle) processRequest(priv *keys.PrivateKey, req request) error {
 	incTx.request = req.Req
 	incTx.tx = tx
 	incTx.backupTx = backupTx
-	incTx.reverifyTx()
+	incTx.reverifyTx(o.Network)
 
-	txSig := priv.Sign(tx.GetSignedPart())
+	txSig := priv.SignHashable(uint32(o.Network), tx)
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/services/oracle/request.go:169]: I don't think so, it uses `Crypto.Sign` directly in C#.

**Comment by roman-khimov** [comment]
> neo-go's implementation of dbft's crypto.PrivateKey interface still uses an old hashing scheme, is it OK?

We're limited by dbft library API here, but I think we're signing everything correctly, at least it worked on our testnet CN in 0.94.0.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**